UnitTests/GitCommands.Tests/AppTitleGeneratorTests.cs

```diff
@@ -1,80 +1,95 @@
-﻿using FluentAssertions;
+﻿using System.Composition;
+using CommonTestUtils.MEF;
+using FluentAssertions;
 using GitCommands;
 using GitCommands.UserRepositoryHistory;
-using NSubstitute;
+using Microsoft.VisualStudio.Composition;
 using NUnit.Framework;
+using ExportAttribute = System.ComponentModel.Composition.ExportAttribute;
+using PartNotDiscoverableAttribute = System.ComponentModel.Composition.PartNotDiscoverableAttribute;
 
 namespace GitCommandsTests
 {
     [TestFixture]
     public class AppTitleGeneratorTests
     {
-        private const string ShortName = "gitextension";
-        private IRepositoryDescriptionProvider _repositoryDescriptionProvider;
-        private AppTitleGenerator _appTitleGenerator;
+        private ExportProvider _mefExportProvider;
 
-        [SetUp]
-        public void Setup()
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
         {
-            _repositoryDescriptionProvider = Substitute.For<IRepositoryDescriptionProvider>();
-            _repositoryDescriptionProvider.Get(Arg.Any<string>()).Returns(ShortName);
+            TestComposition composition = TestComposition.Empty
+                .AddParts(typeof(MockRepositoryDescriptionProvider))
+                .AddParts(typeof(IAppTitleGenerator))
+                .AddParts(typeof(AppTitleGenerator));
 
-            _appTitleGenerator = new AppTitleGenerator(_repositoryDescriptionProvider);
+            _mefExportProvider = composition.ExportProviderFactory.CreateExportProvider();
```

Comment:
❗ Do not reuse a MEF container across tests. The `TestComposition` can be reused, but the result of a call to `CreateExportProvider` should never appear in a OneTimeSetUp method.